src/runtime_src/core/tools/xbmgmt2/ReportPlatform.cpp

```diff
@@ -190,11 +190,17 @@ ReportPlatform::getPropertyTree20202( const xrt_core::device * device,
   f.getBoardInfo(info);
   //create information tree for a device
   pt_platform.put("bdf", xrt_core::query::pcie_bdf::to_string(xrt_core::device_query<xrt_core::query::pcie_bdf>(device)));
-  pt_platform.put("flash_type", xrt_core::device_query<xrt_core::query::flash_type>(device)); 
-  pt_platform.put("hardware.serial_num", info.mSerialNum.empty() ? "N/A" : info.mSerialNum);
+  pt_platform.put("flash_type", xrt_core::device_query<xrt_core::query::flash_type>(device));
+  std::string sn = "N/A";
+  if(info.mSerialNum.empty())
+    sn = xrt_core::device_query<xrt_core::query::hwmon_sdm_serial_num>(device);

```

Comment:
Maybe add this in try/catch? If this node is not available for a non-versal device, it is safe to ignore